### PR TITLE
Restores netrw "gx" to open http URLs under cursor

### LINKED ISSUE #4738: netrw's "gx" functionality doesn't open URLs anymore in macOS
**Describe the bug**
The latest Vim has updated to a newer version of netrw, which seems to have broken the `gx` functionality. In the past, positioning the cursor on an URL and press `gx` would open the URL in a web browser (using the `open` command). The updated netrw would instead try to curl it, which seems wrong.

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Run `vim --clean` (or `gvim --clean`, etc.) in macOS.
2. Enter insert mode and type "https://www.vim.org"
3. Go to normal mode and type `gx`.
4. Observe that Vim doesn't open the URL in a web browser.

**Expected behavior**
Vim should open the URL in a web browser.

**Screenshots**
If applicable, copy/paste the text or add screenshots to help explain your problem.

**Environment (please complete the following information):**
 - Vim version [e.g. 8.1.1755]
 - OS: macOS 10.14 (haven't tested others yet)

**Additional context**
Bug was introduced in 85850f3a5ef9f5a9d22e908ef263de8faa265a95. Originally I got this bug filed against MacVim at https://github.com/macvim-dev/macvim/issues/925 so cross-referencing here.

This is a netrw bug so I'm not sure if this is the correct place to post it, but with the ownership question I'm not 100% sure who owns netrw now and where I should file a bug for it to be fixed and updated to Vim's bundled plugin.

### PR DESCRIPTION
**Pull request by netj**
fixes #4738 in a shoddy way

### CODE REVIEW

File: runtime/autoload/netrw.vim [hunk 1]
```diff
@@ -5489,6 +5489,7 @@ endfun
 
 " ---------------------------------------------------------------------
 " netrw#CheckIfRemote: returns 1 if current file looks like an url, 0 else {{{2
+" FIXME the comment above is no longer true as it takes an optional argument
```
Comment (netj) [inline runtime/autoload/netrw.vim:5492]: https://github.com/vim/vim/blob/8524b091d5c81fe1e1ad8e021e5f36a5c19521d8/runtime/plugin/netrwPlugin.vim#L84

**Comment by netj** [comment]
This currently opens the URL in browser on my MacVim 8.1.1722 in GUI and terminal, but current buffer jumps around to another random one, also losing modifications (need ^I or ^O to get back to it).  So, this is not a complete fix yet and needs more work.

It seems the `netrw#BrowseX()` and a few related functions are too overloaded and needs some refactoring.  I'll try to take a deeper look over the weekend.

**Comment by netj** [comment]
This is not a complete fix yet and needs more work.  With this change I can open the URL under cursor in browser on my MacVim 8.1.1722 in GUI and terminal, but current buffer jumps around to another random one, also losing modifications, and I need `^I` or `^O` to get back to the original buffer. (@cecamp It'd be great if the symptom above rings a bell and you can point me to the right direction!)

It seems the `netrw#BrowseX()` and a few related functions are too overloaded and needs some refactoring.  I'll try to take a deeper look over the weekend.  fwiw I started looking into this around this line in the diff: https://github.com/vim/vim/commit/85850f3a5ef9f5a9d22e908ef263de8faa265a95#diff-532bf738932796a9a60951fb92334bacL84

**Comment by eirnym** [comment]
@netj do you have any luck with refactoring?

**Comment by k-takata** [comment]
This has been already fixed by Netrw v171. Closing.

**Pull request closed without merging**